ui/src/pages/AllStories.tsx

```diff
@@ -34,56 +83,86 @@ function createData(
   };
 }
 
-const StyledAppBar = styled(AppBar)`
-  //styling
-`;
-
 export const AllStories: React.FC = () => {
   const { data: allStories, error } = useSWR<Story[]>("/api/stories");
-  let rows = [];
-
-  // if (allStories===undefined) return;
-  if (allStories) {
-    rows = allStories.map((story, i) =>
-      createData(
-        i,
-        story.title,
-        story.current_city,
-        story.year,
-        story.is_visible,
-        story.author_first_name,
-        story.author_last_name,
-        story.author_country
-      )
-    );
-  }
-
-  const [tabValue, setTabValue] = React.useState(0);
-  const [visibleState, setVisibleState] = React.useState([]);
-  const [tableData, setTableData] = useState(rows);
-
+  const [tabValue, setTabValue] = useState(0);
+  const [visibleState, setVisibleState] = useState([]);
+  const [visibleTableState, setVisibleTableState] = useState([]);
+  const [tableData, setTableData] = useState([]);
+  const [changedVisibility, setChangedVisibility] = useState([]);
```

Comment:
Kk when we have a ton of `useState` let's refactor this into a big `useReducer`.